src/Database/Concerns/ResourceSyncing.php

```diff
@@ -14,15 +14,16 @@ public static function bootResourceSyncing()
     {
         static::saved(function (Syncable $model) {
             /** @var ResourceSyncing $model */
-            $model->triggerSyncEvent();
+            if ($model->isSyncEnabled()) {
+                $model->triggerSyncEvent();
+            }
         });
 
         static::creating(function (self $model) {
-            if (! $model->getAttribute($model->getGlobalIdentifierKeyName()) && app()->bound(UniqueIdentifierGenerator::class)) {
-                $model->setAttribute(
-                    $model->getGlobalIdentifierKeyName(),
-                    app(UniqueIdentifierGenerator::class)->generate($model)
-                );
+            $key = $model->getGlobalIdentifierKeyName();
+
+            if (! $model->getAttribute($key) && app()->bound(UniqueIdentifierGenerator::class) && $model->isSyncEnabled()) {
+                $model->setAttribute($key, app(UniqueIdentifierGenerator::class)->generate($model));
```

Comment:
I can understand the `isSyncEnabled()` change, but what exactly are we doing here? What's the point of the change?

Reply:
I'm sorry, i did not pay attention.
Does not make sense, sync is enabled to set key attribute.
This must be removed.

Reply:
I have removed ```$model->isSyncEnabeld()``` check.

This:
```php
 if (! $model->getAttribute($key) && app()->bound(UniqueIdentifierGenerator::class) && $model->isSyncEnabled()) {
```
should be: 
```php
 if (! $model->getAttribute($key) && app()->bound(UniqueIdentifierGenerator::class)) {
```

Reply:
The code is still changed, so I'm wondering what exactly is the point of the change in this section?

Reply:
It is just a refactoring to turn code more clean. Instead of invoke ` $model->getGlobalIdentifierKeyName()` twice i stored the value in $keyName

Reply:
I see, so there are no logical changes, it's just a refactor?

Reply:
Exacly